middlewares/authorizeRoles.js

```diff
@@ -0,0 +1,57 @@
+const { ROLES } = require("../constants/users");
+
+const VALID_ROLES = [ROLES.SUPERUSER, ROLES.APPOWNER, ROLES.MEMBER];
+
+/**
+ * Check if the user has authorization based on their role.
+ * @param {Array.<String>} allowedRoles - Allowed roles for API consumption.
+ * @param {Object} userRoles - Roles information of the current user.
+ * @returns {Boolean} - Whether the current user is authorized or not.
+ */
+const userHasPermission = (allowedRoles, userRoles) => {
+  let permission = false;
+  Object.keys(userRoles).forEach((role) => {
+    if (userRoles[`${role}`]) {
```

Comment:
On the change that `role` is user defined and there is no type check, a hacker can inject function or abuse prototype chain. To prevent this, eslint gives us an error on `userRoles[role]`. 

Check [here](https://stackoverflow.com/questions/51715616/how-to-fix-codacy-alert-generic-object-injection-sink/55701580#55701580)

``userRoles[`${role}`]`` makes sure that role is parsed into string.